static/js/upload/step1.js

```diff
@@ -0,0 +1,48 @@
+$(document).ready(function() {
```

Comment:
The step1-step4 javascript files were ported from the `upload.js` file, which was becoming too large and messy. Sadly, this PR doesn't manage to show the migration. 

The javascript files don't need to be reviewed in a lot of detail, there are changes, but most of the content is ported from the original file.